File: Array/3042. Count Prefix and Suffix Pairs I/Solution.go
func countPrefixSuffixPairs(words []string) int {
	result := 0
	length := len(words)
	for i := 0; i < length-1; i++ {
		for j := i + 1; j < length; j++ {
			if len(words[i]) <= len(words[j]) {
				if (words[j][:len(words[i])] == words[i]) &&
					(words[j][(len(words[j])-len(words[i])):] == words[i]) {
					result++
				}
			}

		}
	}
	return result
}

import ("strings")
func countPrefixSuffixPairs(words []string) int {
    result:=0;
    length:=len(words);
    for i:=0;i<length-1;i++{
        for j:=i+1;j<length;j++{
            if(strings.HasSuffix(words[j],words[i])&&strings.HasPrefix(words[j],words[i])){
                result++;
            }
        }
    }
    return result;
}

File: Array/3042. Count Prefix and Suffix Pairs I/Solution.c
int countPrefixSuffixPairs(char** words, int wordsSize) {
    int result = 0;
    for (int i = 0; i < wordsSize - 1; i++) {
        for (int j = i + 1; j < wordsSize; j++) {
            // 首先判断words[i]是否<=words[j]
            if (strcmp(words[i], words[j]) <= 0) {
                bool isPrefixAndSuffix = true;
                int prefixLength = strlen(words[i]);
                // 然后判断words[i]是否是words[j]的前缀
                for (int k = 0; k < prefixLength; k++) {
                    if (words[i][k] != words[j][k]) {
                        isPrefixAndSuffix = false;
                        break;
                    }
                }
                // 最后判断words[i]是否是words[j]的后缀
                if (isPrefixAndSuffix) {
                    int suffixLength = strlen(words[j]) - prefixLength;
                    for (int k = 0; k < prefixLength; k++) {
                        if (words[i][k] != words[j][suffixLength + k]) {
                            isPrefixAndSuffix = false;
                            break;
                        }
                    }
                }
                if (isPrefixAndSuffix) {
                    result++;
                }
            }
        }
    }
    return result;
}